src/main/java/twistedgate/immersiveposts/client/model/PostBaseModel.java

```diff
@@ -196,34 +185,23 @@ private void build(Key key){
 		}
 		
 		@Override
-		public List<BakedQuad> getQuads(@Nullable BlockState state, @Nullable Direction side, @Nullable RandomSource rand, @Nullable ModelData extraData, RenderType layer){
+		public @NotNull List<BakedQuad> getQuads(@Nullable BlockState state, @Nullable Direction side, @Nullable RandomSource rand, @Nullable ModelData extraData, RenderType layer){
 			return this.quads.get(side == null ? (this.quads.size() - 1) : side.get3DDataValue());
 		}
 	}
-	
-	private static class Key{
-		final BlockState state;
-		final Direction facing;
-		
-		public Key(BlockState state, Direction facing){
-			this.state = state;
-			this.facing = facing;
-		}
-		
+
+	private record Key(BlockState state, Direction facing) {
+
 		@Override
-		public boolean equals(Object obj){
-			if(this == obj){
-				return true;
-			}else if(obj != null && obj instanceof Key other){
-				return this.state.equals(other.state) && this.facing == other.facing;
-			}
-			
+			public boolean equals(Object obj) {
+				if (this == obj) {
+					return true;
+				} else if (obj instanceof Key other) {
+					return this.state.equals(other.state) && this.facing == other.facing;
+				}
+
 			return false;
-		}
-		
-		@Override
-		public int hashCode(){
-			return 31 * this.state.hashCode() + Objects.hash(this.facing);
-		}
+			}
```

Comment:
This just sent me.

Calling loads of my stuff useless while adding useless boiler-plate code into a Record whos whole point is to not having to write Boiler-Plate

Reply:
As I said, I was not referring to them as a "bad code", but, not called method are useless (and there were some out of the API context